controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -342,61 +354,70 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
+        return getTransactionNameFromId(qualifiedName, txId);
+    }
+
     public CompletableFuture<TxnStatus> commitTransaction(final String scope,
                                                           final String stream,
-                                                          final long segmentId,
+                                                          final long targetSegmentId,
+                                                          final long sourceSegmentId,
                                                           final UUID txId,
                                                           final HostControllerStore hostControllerStore,
                                                           final ConnectionFactory clientCF, String delegationToken) {
-
-        final Controller.NodeUri uri = getSegmentUri(scope, stream, segmentId, hostControllerStore);
-        final String qualifiedName = StreamSegmentNameUtils.getQualifiedStreamSegmentName(scope, stream, segmentId);
+        Preconditions.checkArgument(getSegmentNumber(targetSegmentId) == getSegmentNumber(sourceSegmentId));
+        final Controller.NodeUri uri = getSegmentUri(scope, stream, sourceSegmentId, hostControllerStore);
+        final String qualifiedNameTarget = getQualifiedStreamSegmentName(scope, stream, targetSegmentId);
+        final String transactionName = getTransactionName(scope, stream, sourceSegmentId, txId);
         final CompletableFuture<TxnStatus> result = new CompletableFuture<>();
-        final WireCommandType type = WireCommandType.COMMIT_TRANSACTION;
+        final WireCommandType type = WireCommandType.MERGE_SEGMENTS;
         final FailingReplyProcessor replyProcessor = new FailingReplyProcessor() {
 
             @Override
             public void connectionDropped() {
-                log.warn("commitTransaction {} connection dropped", qualifiedName);
+                log.warn("commitTransaction {} connection dropped", transactionName);
                 result.completeExceptionally(
                         new WireCommandFailedException(type, WireCommandFailedException.Reason.ConnectionDropped));
             }
 
             @Override
             public void wrongHost(WireCommands.WrongHost wrongHost) {
-                log.warn("commitTransaction {} wrongHost", qualifiedName);
+                log.warn("commitTransaction {} wrongHost", transactionName);
                 result.completeExceptionally(
                         new WireCommandFailedException(type, WireCommandFailedException.Reason.UnknownHost));
             }
 
             @Override
-            public void transactionCommitted(WireCommands.TransactionCommitted transactionCommitted) {
-                log.debug("commitTransaction {} TransactionCommitted", qualifiedName);
+            public void segmentsMerged(WireCommands.SegmentsMerged segmentsMerged) {
+                log.debug("commitTransaction {} TransactionCommitted", transactionName);
```

Comment:
change log message to match new method name

Reply:
The CommitTransaction is replaced by WireCommand.MergeSegment. 
So the context in which mergeSegment is called is for commitTxn. Hence the log message treats segmentsMerged as a sign of transcationCommitted.